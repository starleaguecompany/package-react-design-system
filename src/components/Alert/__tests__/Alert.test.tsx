import * as React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Menu } from '@starleaguecompany/react-icons';
import { Alert } from '../../../index';

describe('Alert', () => {
  test('should render correctly with no props', () => {
    render(<Alert />);

    expect(screen.getByTestId('Alert')).toMatchSnapshot();
    expect(screen.getByTestId('Alert')).toHaveClass('container color-dark', { exact: true });
  });

  test('should render correctly with attributes', () => {
    render(<Alert id="test-id" className="test-class" />);

    expect(screen.getByTestId('Alert')).toMatchSnapshot();
    expect(screen.getByTestId('Alert')).toHaveAttribute('id', 'test-id');
    expect(screen.getByTestId('Alert')).toHaveClass('test-class');
  });

  test('should render correctly with color property', () => {
    render(<Alert color="orange" />);

    expect(screen.getByTestId('Alert')).toMatchSnapshot();
    expect(screen.getByTestId('Alert')).toHaveClass('color-orange');
  });

  test('should render correctly with title prop', () => {
    render(<Alert title="Testing Alert title" />);

    expect(screen.getByTestId('Alert')).toMatchSnapshot();
    expect(screen.getByTestId('Alert')).toContainElement(screen.getByText('Testing Alert title'));
  });

  test('should render correctly with title and subtitle prop', () => {
    render(<Alert title="Testing Alert title" subtitle="Testing Alert subtitle" />);

    expect(screen.getByTestId('Alert')).toMatchSnapshot();
    expect(screen.getByTestId('Alert')).toContainElement(screen.getByText('Testing Alert title'));
    expect(screen.getByTestId('Alert')).toContainElement(screen.getByText('Testing Alert subtitle'));
  });

  test('should render correctly with icon prop', () => {
    render(<Alert icon={<Menu />} />);

    expect(screen.getByTestId('Alert')).toMatchSnapshot();
    expect(screen.getByTestId('Alert')).toContainElement(screen.getByTestId('Menu'));
    expect(screen.getByTestId('Menu')).toBeVisible();
  });

  test('should render correctly with children', () => {
    render(
      <Alert>
        <Menu />
      </Alert>
    );

    expect(screen.getByTestId('Alert')).toMatchSnapshot();
    expect(screen.getByTestId('Alert')).toContainElement(screen.getByTestId('Menu'));
    expect(screen.getByTestId('Menu')).toBeVisible();
  });

  test('should render correctly with footer prop', () => {
    render(<Alert footer={<Menu />} />);

    expect(screen.getByTestId('Alert')).toMatchSnapshot();
    expect(screen.getByTestId('Alert')).toContainElement(screen.getByTestId('Menu'));
    expect(screen.getByTestId('Menu')).toBeVisible();
  });

  test('should render correctly with closeable prop', () => {
    render(<Alert title="Testing Alert title" closable />);

    expect(screen.getByTestId('Alert')).toMatchSnapshot();
    expect(screen.getByTestId('Alert')).toContainElement(screen.getByTestId('Cross'));
    expect(screen.getByTestId('Cross')).toBeVisible();
  });

  test('should execute callback on close button click', () => {
    const closeCallback = jest.fn();
    render(<Alert closable onClose={closeCallback} />);

    expect(screen.getByTestId('Alert')).toMatchSnapshot();
    userEvent.click(screen.getByTestId('Cross'));
    expect(closeCallback).toBeCalledTimes(1);
    userEvent.click(screen.getByTestId('Alert'));
    expect(closeCallback).toBeCalledTimes(1);
  });
});
