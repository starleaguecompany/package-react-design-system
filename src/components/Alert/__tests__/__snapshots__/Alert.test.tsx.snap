// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should execute callback on close button click 1`] = `
<div
  class="container color-dark"
  data-qa="Alert"
>
  <div
    class="header container direction-horizontal align-center"
    data-qa="Space"
    style="column-gap: 12px; row-gap: 12px;"
  >
    <div
      class="titleWrapper container direction-vertical"
      data-qa="Space"
    />
    <span
      class="close container size-20 shape-round"
      data-qa="Icon"
    >
      <svg
        data-qa="Cross"
        fill="none"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#clip0)"
        >
          <path
            d="m10 10-4.95 4.95m0-9.9L10 10 5.05 5.05ZM10 10l4.95 4.95L10 10Zm0 0 4.95-4.95L10 10Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
          <path
            d="M79-22H-51a5 5 0 0 0-5 5v54a5 5 0 0 0 5 5H79a5 5 0 0 0 5-5v-54a5 5 0 0 0-5-5Z"
            stroke="currentColor"
            stroke-dasharray="10 5"
          />
        </g>
        <defs>
          <clippath
            id="clip0"
          >
            <path
              d="M0 0h20v20H0z"
              fill="currentColor"
            />
          </clippath>
        </defs>
      </svg>
    </span>
  </div>
</div>
`;

exports[`Alert should render correctly with attributes 1`] = `
<div
  class="test-class container color-dark"
  data-qa="Alert"
  id="test-id"
/>
`;

exports[`Alert should render correctly with children 1`] = `
<div
  class="container color-dark"
  data-qa="Alert"
>
  <div
    class="content"
  >
    <svg
      data-qa="Menu"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 5h12M4 10h12M4 15h12"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.5"
      />
    </svg>
  </div>
</div>
`;

exports[`Alert should render correctly with closeable prop 1`] = `
<div
  class="container color-dark"
  data-qa="Alert"
>
  <div
    class="header container direction-horizontal align-center"
    data-qa="Space"
    style="column-gap: 12px; row-gap: 12px;"
  >
    <div
      class="titleWrapper container direction-vertical"
      data-qa="Space"
    >
      <div
        class="title"
      >
        Testing Alert title
      </div>
    </div>
    <span
      class="close container size-20 shape-round"
      data-qa="Icon"
    >
      <svg
        data-qa="Cross"
        fill="none"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          clip-path="url(#clip0)"
        >
          <path
            d="m10 10-4.95 4.95m0-9.9L10 10 5.05 5.05ZM10 10l4.95 4.95L10 10Zm0 0 4.95-4.95L10 10Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
          <path
            d="M79-22H-51a5 5 0 0 0-5 5v54a5 5 0 0 0 5 5H79a5 5 0 0 0 5-5v-54a5 5 0 0 0-5-5Z"
            stroke="currentColor"
            stroke-dasharray="10 5"
          />
        </g>
        <defs>
          <clippath
            id="clip0"
          >
            <path
              d="M0 0h20v20H0z"
              fill="currentColor"
            />
          </clippath>
        </defs>
      </svg>
    </span>
  </div>
</div>
`;

exports[`Alert should render correctly with color property 1`] = `
<div
  class="container color-orange"
  data-qa="Alert"
/>
`;

exports[`Alert should render correctly with footer prop 1`] = `
<div
  class="container color-dark"
  data-qa="Alert"
>
  <div
    class="footer container direction-horizontal justify-stretch"
    data-qa="Space"
    style="column-gap: 12px; row-gap: 12px;"
  >
    <svg
      data-qa="Menu"
      fill="none"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 5h12M4 10h12M4 15h12"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="1.5"
      />
    </svg>
  </div>
</div>
`;

exports[`Alert should render correctly with icon prop 1`] = `
<div
  class="container color-dark"
  data-qa="Alert"
>
  <div
    class="header container direction-horizontal align-center"
    data-qa="Space"
    style="column-gap: 12px; row-gap: 12px;"
  >
    <div
      class="icon"
    >
      <svg
        data-qa="Menu"
        fill="none"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 5h12M4 10h12M4 15h12"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </div>
    <div
      class="titleWrapper container direction-vertical"
      data-qa="Space"
    />
  </div>
</div>
`;

exports[`Alert should render correctly with no props 1`] = `
<div
  class="container color-dark"
  data-qa="Alert"
/>
`;

exports[`Alert should render correctly with title and subtitle prop 1`] = `
<div
  class="container color-dark"
  data-qa="Alert"
>
  <div
    class="header container direction-horizontal align-center"
    data-qa="Space"
    style="column-gap: 12px; row-gap: 12px;"
  >
    <div
      class="titleWrapper container direction-vertical"
      data-qa="Space"
    >
      <div
        class="title"
      >
        Testing Alert title
      </div>
      <div
        class="subtitle"
      >
        Testing Alert subtitle
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert should render correctly with title prop 1`] = `
<div
  class="container color-dark"
  data-qa="Alert"
>
  <div
    class="header container direction-horizontal align-center"
    data-qa="Space"
    style="column-gap: 12px; row-gap: 12px;"
  >
    <div
      class="titleWrapper container direction-vertical"
      data-qa="Space"
    >
      <div
        class="title"
      >
        Testing Alert title
      </div>
    </div>
  </div>
</div>
`;
